processor/transformprocessor/internal/traces/expression.go

```diff
@@ -0,0 +1,81 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type exprFunc func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{}
+
+// getter allows reading a value while processing traces. Note that data is not necessarily read from input
+// telemetry but may be a literal value or a function invocation.
+type getter interface {
+	get(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{}
+}
+
+// getSetter allows reading or writing a value to trace data.
+type getSetter interface {
+	getter
+	set(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
+}
```

Comment:
Do we not need just a "setter"? Always need a getter/setter?

Reply:
All cases I've seen in your code are actually you either need a Setter or a Getter, never both... I think that was my question, do we better just have 2 interfaces Setter/Getter and some implementations my have both?

Reply:
So I would:
```golang
// getter allows reading a value while processing traces. Note that data is not necessarily read from input
// telemetry but may be a literal value or a function invocation.
type getter interface {
	get(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{}
}

// setter allows writing a value to trace data.
type setter interface {
	set(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
}

// getSetter allows reading or writing a value to trace data.
type getSetter interface {
	getter
	setter
}
```